src/internal/kopia/upload_test.go

```diff
@@ -183,16 +183,22 @@ func expectDirs(
 ) {
 	t.Helper()
 
-	if exactly {
-		require.Len(t, entries, len(dirs))
+	ents := make([]string, 0, len(entries))
+	for _, e := range entries {
+		ents = append(ents, e.Name())
 	}
 
-	names := make([]string, 0, len(entries))
-	for _, e := range entries {
-		names = append(names, e.Name())
+	dd, err := decodeElements(dirs...)
+	require.NoError(t, err, clues.ToCore(err))
+
+	de, err := decodeElements(ents...)
+	require.NoError(t, err, clues.ToCore(err))
+
+	if exactly {
+		require.Lenf(t, entries, len(dirs), "expected %+v\ngot %+v", dd, de)
 	}
 
-	assert.Subset(t, names, dirs)
+	assert.Subsetf(t, dirs, ents, "expected %+v\ngot %+v", dd, de)
```

Comment:
nit: would be nice to say something about expecting "at least" since this is a subset check